### ISSUE TITLE
no-undef does not check Node.js scope for variables used in JSX

### ISSUE THREAD

**Issue by yannickcr** [eslint/eslint#2093]
no-undef does not check Node.js scope for variables used in JSX

test.js

```javascript
var App = require('App');

module.exports = <App />
```

.eslintrc

```yaml
env:
    node: true
ecmaFeatures:
    jsx: true
rules:
    no-undef: 2
```

Output:

```bash
test.js
  3:18  error  'App' is not defined                                                 no-undef

✖ 1 problem (1 error, 0 warnings)
```

**Comment by srigi** [comment]
+1, have same problem

**Comment by nzakas** [comment]
Can you generalize this to something that isn't related to JSX/React? Right now, this is the correct behavior for ESLint without any plugins.

**Comment by srigi** [comment]
This looks like bug in [babel-eslint](https://github.com/babel/babel-eslint/issues/15). Non-JSX code checking works well.

**Comment by yannickcr** [comment]
@nzakas This is totally related to JSX since the problem is in the `checkIdentifierInJSX` function.

I think this is the same problem as #2090 or #2064, the Node.js scope is not checked.

**Comment by nzakas** [comment]
Ah, I thought that had been removed in the last release. We really shouldn't be checking JSX tag names in ESLint, as that's part of React semantics. So I think what we should be doing here is removing that functionality from `no-undef`.

**Comment by yannickcr** [comment]
Sounds good to me. This way `no-undef` will stop reporting some errors on JSX tag names and I can add a JSX specific rule for this in my plugin.

**Comment by nzakas** [comment]
:+1:

**Comment by nzakas** [comment]
Working on this.
